### PR TITLE
Enable SSL client authentication

### PR DESCRIPTION
**Pull request by rbtcollins**
Signed-off-by: Robert Collins <robertc@vmware.com>

### CODE REVIEW

**Comment by rbtcollins** [comment]
Warning: I haven't tested this yet, it may need further work, but it does seem to be basically this simple.

File: jaeger-spark-dependencies-cassandra/src/main/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJob.java [hunk 1]
```diff
@@ -78,6 +78,12 @@ public static Builder builder() {
           System.getProperty("javax.net.ssl.trustStorePassword", ""));
       sparkProperties.put("spark.cassandra.connection.ssl.trustStore.path",
           System.getProperty("javax.net.ssl.trustStore", ""));
+      sparkProperties.put("spark.cassandra.connection.ssl.clientAuth.enabled",
+          Utils.getEnv("CASSANDRA_SSL_KEYSTORE", "").isEmpty() ? "false" : "true");
+      sparkProperties.put("spark.cassandra.connection.ssl.keyStore.path",
```
Comment (pavolloffay) [inline jaeger-spark-dependencies-cassandra/src/main/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJob.java]: Shouldn't this be configurable? The default is None so why setting it to an empty string?

https://github.com/datastax/spark-cassandra-connector/blob/master/doc/reference.md

**Comment by rbtcollins** [comment]
Shouldn't this be configurable? The default is None so why setting it to
an empty string?


Commenting from phone, but if this is the line I think it is, passing none
in raises an exception, regardless of the default value.

**Comment by pavolloffay** [comment]
@rbtcollins did you test this? As your previous comment says no.

**Comment by rbtcollins** [comment]
I haven't yet done full functional testing of it no; thats why I mentioned it. :). I was unsure if the change was related to the master test failures previously; I haven't gotten back to it just yet.

**Comment by pavolloffay** [comment]
done in #26

**Pull request closed without merging**